typescript/cli/src/context.ts

```diff
@@ -3,26 +3,38 @@ import { ethers } from 'ethers';
 import {
   ChainMap,
   ChainMetadata,
+  ChainName,
   HyperlaneContractsMap,
   MultiProvider,
   chainMetadata,
   hyperlaneEnvironments,
 } from '@hyperlane-xyz/sdk';
-import { objMerge } from '@hyperlane-xyz/utils';
+import { objFilter, objMap, objMerge } from '@hyperlane-xyz/utils';
 
 import { readChainConfigsIfExists } from './config/chain.js';
 import { keyToSigner } from './utils/keys.js';
 
-export const sdkContractAddressesMap = {
+export const sdkContractAddressesMap: HyperlaneContractsMap<any> = {
   ...hyperlaneEnvironments.testnet,
   ...hyperlaneEnvironments.mainnet,
 };
 
 export function getMergedContractAddresses(
   artifacts?: HyperlaneContractsMap<any>,
+  chains?: ChainName[],
 ) {
+  // if chains include non sdkContractAddressesMap chains, don't recover interchainGasPaymaster
+  let sdkContractsAddressesToRecover = sdkContractAddressesMap;
+  if (chains?.some((chain) => !(chain in sdkContractAddressesMap))) {
```

Comment:
Nit: the `in` keyword has some undesirable pitfalls. It's recommended to use `hasOwnProperty` or `Object.keys` instead: https://www.quora.com/Why-is-extending-built-in-JavaScript-types-evil